No need to emit additonalscript event on public pages

rullzer opened the pull request:
There already is a separate event for this. This will make it possible
to only inject code with the logged in one on default rendered pages.

rullzer committed:
No need to emit additonalscript event on public pages

There already is a separate event for this. This will make it possible
to only inject code with the logged in one on default rendered pages.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

nickvergessen commented:
Could we exclude `/login/` too? That is the two factor pages, where the user is already logged in, but not done with authentication :(
https://github.com/nextcloud/notifications/blob/master/lib/AppInfo/Application.php#L75

rullzer commented:
@nickvergessen ha good point. Lets see how we can do that. Maybe with a separate interface for those pages? I'd like to avoid relying on url magic.

nickvergessen commented:
> I'd like to avoid relying on url magic.

:scream: No 🔮 for you?
Yeah sounds sane

rullzer committed:
Add StandaloneTemplateResponse

This can be used by pages that do not have the full Nextcloud UI.
So notifications etc do not load there.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
Ok updated. Lets see if CI is happy

MorrisJobke commented on lib/public/AppFramework/Http/StandAloneTemplateResponse.php:
```diff
@@ -0,0 +1,37 @@
+ *
+ * @since 16.0.0
+ */
+class StandAloneTemplateResponse extends TemplateResponse {
```
Isn't "standalone" one word? Then it would be `StandaloneTemplateResponse`, or not?

rullzer replied:
yes I guess you are right. Will fix

rullzer replied:
Done

rullzer commented:
@nickvergessen @MorrisJobke please have another look.

MorrisJobke commented on lib/private/AppFramework/Middleware/AdditionalScriptsMiddleware.php:
```diff
@@ -42,6 +44,14 @@ public function __construct(EventDispatcherInterface $dispatcher, IUserSession $
+		 * There is no need to emit these signals on a public share page
+		 * There is a separate event for that already
+		 */
+		if ($controller instanceof PublicShareController || $controller instanceof StandaloneTemplateResponse) {
```
```suggestion
		if ($controller instanceof PublicShareController || $response instanceof StandaloneTemplateResponse) {
```

Or did I miss something?

rullzer replied:
It of course helps if I push the latest changes :S sorry.
Anyway fixed and with tests

MorrisJobke commented on lib/public/AppFramework/Http/StandaloneTemplateResponse.php:
```diff
@@ -0,0 +1,37 @@
+ * A template response that does not emit the loadAdditionalScripts events.
+ *
+ * This is useful for pages that are authenticated but do not yet show the
+ * full nextcloud UI. Like the 2FA page, or the grant page in the login flow.
```
Doesn't work on the normal login page.

rullzer replied:
The login page is not authenticated right?

rullzer replied:
Ah you mean that it should emit the regular loadAdditionalScripts on the login page?

MorrisJobke replied:
> Ah you mean that it should emit the regular loadAdditionalScripts on the login page?

Correct :)

rullzer replied:
fixed

MorrisJobke reviewed (approved):
Tested and works 👍
